tvplayer/src/main/java/com/kaltura/tvplayer/KalturaPlayer.java

```diff
@@ -430,6 +440,30 @@ public void setPlayerView(int playerWidth, int playerHeight) {
     }
 
     public void setMedia(@NonNull PKMediaEntry mediaEntry) {
+        applyMediaEntryInterceptors(mediaEntry, () ->
+                mainHandler.post(() -> {
+                    tokenResolver.update(mediaEntry, getKS());
+
+                    if (externalSubtitles != null) {
+                        if (mediaEntry.getExternalSubtitleList() == null) {
+                            mediaEntry.setExternalSubtitleList(externalSubtitles);
+                        } else {
+                            mediaEntry.getExternalSubtitleList().addAll(externalSubtitles);
+                        }
+                    }
+
+                    if (preload) {
+                        this.mediaEntry = mediaEntry;
+                        this.prepareState = PrepareState.not_prepared;
+                        PKPluginConfigs combinedPluginConfigs = setupPluginsConfiguration();
+                        updateKalturaPluginConfigs(combinedPluginConfigs);
+                        prepare();
+                    }
+                }));
+
```

Comment:
can't we just call `setMediaInternal` here?

Reply:
this will also make `setMedia` an async method, can it break anything for app?

Reply:
done